pkg/apis/core/v1alpha1/helper/shootstate_lister.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helper
+
+import (
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+)
+
+// GetExtensionResourceStateList returns a list of ExtensionResourceStates which can be modified.
+func GetExtensionResourceStateList(shootState *gardencorev1alpha1.ShootState) ExtensionResourceStateList {
+	return ExtensionResourceStateList(shootState.Spec.Extensions)
+}
+
+// SetExtensionResourceStateList sets the provided ExtensionResourceStateList to the ShootState.
+func SetExtensionResourceStateList(shootState *gardencorev1alpha1.ShootState, extensionsList ExtensionResourceStateList) {
```

Comment:
Same here - why do we need such a function if can directly do what you do in line 29 instead of invoking it?